src/api/uploads/MultiputPart.js

```diff
@@ -34,30 +51,264 @@ class MultiputPart extends Base {
      * @param {number} index - 0-based index of this part in array of all parts
      * @param {number} offset - Starting byte offset of this part's range
      * @param {number} size - Size of this part in bytes
+     * @param {number} sessionId
+     * @param {Object} sessionEndpoints
+     * @param {MultiputConfig} config
+     * @param {Function} getPartsState
+     * @param {Function} getNumPartsUploading
+     * @param {Function} [onSuccess]
+     * @param {Function} [onProgress]
+     * @param {Function} [onError]
      * @return {void}
      */
-    constructor(options: Object, index: number, offset: number, size: number): void {
-        super(options);
+    constructor(
+        options: Object,
+        index: number,
+        offset: number,
+        size: number,
+        sessionId: string,
+        sessionEndpoints: Object,
+        config: MultiputConfig,
+        getPartsState: Function,
+        getNumPartsUploading: Function,
+        onSuccess?: Function,
+        onProgress?: Function,
+        onError?: Function
+    ): void {
+        super(options, sessionEndpoints, config);
 
         this.index = index;
         this.numDigestRetriesPerformed = 0;
         this.numUploadRetriesPerformed = 0;
         this.offset = offset;
-        this.sha256 = null;
         this.size = size;
         this.state = PART_STATE_NOT_STARTED;
         this.timing = {};
         this.uploadedBytes = 0;
+        this.data = {};
+        this.uploadUrl = `${this.uploadHost}/api/2.0/files/upload_sessions/${sessionId}`;
+        this.config = config;
+        this.rangeEnd = offset + size - 1;
+        this.onSuccess = onSuccess || noop;
+        this.onError = onError || noop;
+        this.onProgress = onProgress || noop;
+        this.getPartsState = getPartsState;
+        this.getNumPartsUploading = getNumPartsUploading;
     }
 
+    stringify = () =>
+        JSON.stringify({
+            index: this.index,
+            offset: this.offset,
+            size: this.size,
+            state: this.state,
+            uploadedBytes: this.uploadedBytes,
+            numUploadRetriesPerformed: this.numUploadRetriesPerformed,
+            numDigestRetriesPerformed: this.numDigestRetriesPerformed,
+            sha256: this.sha256,
+            xhr: this.xhr.xhr,
```

Comment:
Not sure what this would result in.
Might just be `{}`

Reply:
Add a `toJSON()` method in XHR.js if you want more useful info. Then calling JSON.stringify() will automatically use that method.

Reply:
This whole method can be the toJSON method for this class too

Reply:
will do

Reply:
looks like currently in the webapp `xhr` field printed is always null or {}, removing that from being printed